### PR TITLE
Add grayscale to palette, add as button color

### PR DESCRIPTION
**Pull request by gigxz**
## Description

Issue: https://github.com/open-path/Green-River/issues/7130
Chromatic: [Link](https://66e877f06b978106d75d04bc-jyehnncswm.chromatic.com/?path=/story/components-elements-button--all-variants)

This PR is a refactor of "gray" buttons in the app, to match the hmis design library.

- Add `grayscale` color palette per design library
- Replace usage of old `gray` button variant color `color=grayscale variant=contained`
- Replace usage of old `transparent` button variant with `color=grayscale variant=text`
- Update TODOs
- Add button variant/color stories
- Remove default white background from `outlined` buttons per figma change (see https://github.com/open-path/Green-River/issues/7130#issuecomment-2560383390). This allows them to be used on alerts correctly.

## Type of change
- [x] Code clean-up / housekeeping

## Checklist before requesting review
- [x] I have performed a self-review of my code
- [x] I have run the code that is being changed under ideal conditions, and it doesn't fail
- [x] My code includes comments and/or descriptive variable names to help other engineers understand the intent (or not applicable)
- [x] My code follows the style guidelines of this project (eslint)
- [x] I have updated the documentation (or not applicable)
- [x] If it's not obvious how to test this change, I have provided testing instructions in this PR or the related issue

### CODE REVIEW

File: src/config/theme.ts [hunk 1]
```diff
@@ -126,6 +139,14 @@ export const baseThemeDef: ThemeOptions = {
...
     links: '#1976D2',
     activeStatus: '#75559F',
+    grayscale: {
+      main: '#6E6E6E',
+      dark: '#4D4D4D',
+      contrastText: '#fff',
+      p30: alpha('#6E6E6E', 0.3),
+      p12: alpha('#6E6E6E', 0.12),
+      p08: alpha('#6E6E6E', 0.08),
+    },
```
Comment (gigxz) [inline src/config/theme.ts:149]: Review w/ Bobby - this is how things appear specified in figma, so trying to match that

File: src/config/theme.ts [hunk 2]
```diff
@@ -418,10 +440,31 @@ const createThemeOptions = (theme: Theme) => ({
...
+          style: theme.unstable_sx({
+            backgroundColor: theme.palette.grayscale.p08,
+          }),
+        },
+        {
+          props: { variant: 'outlined', color: 'grayscale' },
+          style: theme.unstable_sx({
+            borderColor: theme.palette.grayscale.p30,
+            '&:hover': {
+              borderColor: theme.palette.grayscale.p30,
```
Comment (gigxz) [inline src/config/theme.ts:465]: Looks like new designs have this across the board (outlined buttons border color shouldn't change on hover), see if there is a way to configure that more generally

File: src/config/Color.mdx [hunk 3]
```diff
@@ -33,13 +33,24 @@ export function extractColors(paletteObject, prefix = '') {
...
+
+return entries.map(([k, v]) => {
+if (isColor(v)) {
+return <ColorItem key={k} title={k} colors={{ [k]: v }} subtitle='' />;
+} else if (typeof v === 'object') {
+const colors = extractColors(v);
+return <ColorItem key={k} title={k} colors={colors} subtitle='' />;
+}
+return null;
+});
```
Comment (ttoomey) [inline src/config/Color.mdx]: I'm not very familiar with mdx but the indentation seems off here

File: src/config/theme.ts [hunk 4]
```diff
@@ -55,17 +48,25 @@ declare module '@mui/material/styles' {
       icon?: string;
     };
   }
+  interface GrayscalePaletteColor
+    extends PaletteColor,
+      Pick<Color, 300 | 200 | 100> {
+    tint: string;
+  }
```
Comment (ttoomey) [inline src/config/theme.ts:55]: add for IconButton too? Then you don't need to hard cast to IconButtonProps (in DeleteMutationButton)
```
declare module '@mui/material/IconButton' {
  interface IconButtonPropsColorOverrides {
    grayscale: true;
  }
}
```

**Review by ttoomey — APPROVED** [review]
Looks great!

File: src/config/Color.mdx (hunk 3 above)
Comment (gigxz) [reply src/config/Color.mdx]: This was `prettier`s doing- I added mdx to prettierignore

**Pull request merged**